Hogajama/hogajama-domain/src/main/java/com/gepardec/hogarama/domain/unitmanagement/service/ActorService.java

```diff
@@ -0,0 +1,52 @@
+package com.gepardec.hogarama.domain.unitmanagement.service;
+
+import com.gepardec.hogarama.domain.exception.TechnicalException;
+import com.gepardec.hogarama.domain.unitmanagement.context.UserContext;
+import com.gepardec.hogarama.domain.unitmanagement.dao.ActorDAO;
+import com.gepardec.hogarama.domain.unitmanagement.entity.Actor;
+import com.gepardec.hogarama.domain.unitmanagement.entity.Unit;
+
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ActorService {
+
+    @Inject
+    private ActorDAO dao;
+
+    @Inject
+    private UserContext userContext;
+
+    public void createActor(Actor actor) {
+        verifyUnitBelongsToOwner(actor.getUnit());
+        dao.save(actor);
+    }
+
+    public void deleteActor(Long actorId) {
+        Actor actor = this.dao.getById(actorId)
+                .orElseThrow(() -> new NotFoundException(String.format("Actor with id [%d] not found", actorId)));
+
+        dao.delete(actor);
+    }
+
+    public void updateActor(Actor actor) {
+        verifyUnitBelongsToOwner(actor.getUnit());
+        dao.update(actor);
+    }
+
+    public List<Actor> getAllActorsForOwner() {
+        return dao.getAllActorsForOwner(userContext.getOwner());
+    }
+
+    private void verifyUnitBelongsToOwner(Unit unit) {
```

Comment:
The same code is used within Sensor- and ActorService, would it make sense to move it to the OwnerService class?

Reply:
If you look where this called you find e.g:
```
    public void createActor(Actor actor) {
        verifyUnitBelongsToOwner(actor.getUnit());
        dao.save(actor);
    }
```
What is meant is: `currentUser.hasPermissionToManage(actor)`, implemented by `actor.isOwendBy(currentUser)`. The fact that the owner is determined by the unit is implementation detail and should be hidden in the domain model, not in the service layer. (This is a little nitpicking, but I think it helps to discuss such things.)

When it gets down to:
```
    private boolean unitBelongsToOwner(Unit unit) {
        return userContext.getOwner().getUnitList().stream().map(Unit::getId).collect(Collectors.toSet()).contains(unit.getId());
    }
```
I ask why not navigate the other way, like: `unit.getOwner().equals(userContext.getOwner())`
With an interface like `Owend` we could push most of the logic to the domain model. I usually aim for a rich domain model instead of an anemic domain model. 

`OwnerService` registers and saves new Owners. In my view this is not a good place for such logic.